pytorch_lightning/accelerators/ddp_backend.py

```diff
@@ -123,14 +139,116 @@ def __ddp_script_mode_setup(self):
 
         self.task_idx = 0
 
+        # wait for all the procs to start
+        sleep(2)
+
     def train(self):
         model = self.trainer.model
         if self.mode == 'ddp':
-            results = self.ddp_train_tmp(process_idx=self.task_idx, mp_queue=None, model=model, is_master=True)
-            del os.environ['WORLD_SIZE']
+            results = self.ddp_train(process_idx=self.task_idx, mp_queue=None, model=model, is_master=True)
+            if 'WORLD_SIZE' in os.environ:
+                del os.environ['WORLD_SIZE']
             return results
         else:
-            self.ddp_train_tmp(process_idx=self.task_idx, mp_queue=None, model=model)
+            return self.ddp_train(process_idx=self.task_idx, mp_queue=None, model=model)
+
+    def ddp_train(self, process_idx, mp_queue, model, is_master=False, proc_offset=0):
+        """
+        Entry point for ddp
+        Args:
+            process_idx:
+            mp_queue: multiprocessing queue
+            model:
+            is_master:
+            proc_offset:
+        Returns:
+        """
+        # offset the process id if requested
+        process_idx = process_idx + proc_offset
+
+        # show progressbar only on progress_rank 0
+        if (self.trainer.node_rank != 0 or process_idx != 0) and self.trainer.progress_bar_callback is not None:
+            self.trainer.progress_bar_callback.disable()
+
+        # determine which process we are and world size
+        self.set_world_ranks(process_idx)
+
+        # set warning rank
+        rank_zero_only.rank = self.trainer.global_rank
+
+        # set up server using proc 0's ip address
+        # try to init for 20 times at max in case ports are taken
+        # where to store ip_table
+        model.trainer = self.trainer
+        model.init_ddp_connection(
+            self.trainer.global_rank,
+            self.trainer.world_size,
+            self.trainer.is_slurm_managing_tasks
+        )
+
+        # call setup after the ddp process has connected
+        self.trainer.call_setup_hook(model)
+
+        # on world_size=0 let everyone know training is starting
+        if self.trainer.is_global_zero and not torch.distributed.is_initialized():
```

Comment:
just thinking, but is_global_zero should be part of DDPBackend IMO, since this is only needed for this